ravenframework/Optimizers/Optimizer.py

```diff
@@ -208,7 +208,8 @@ def _localGenerateAssembler(self, initDict):
     self.assemblerDict['DataObjects'] = []
     self.assemblerDict['Distributions'] = []
     self.assemblerDict['Functions'] = []
-    for mainClass in ['DataObjects', 'Distributions', 'Functions']:
+    self.assemblerDict['Files'] = []
+    for mainClass in ['DataObjects', 'Distributions', 'Functions', 'Files']:
```

Comment:
maybe define a validAssembler list which can be used in multiple places.

Reply:
added